lib/dialects/abstract/query.js

```diff
@@ -514,7 +514,19 @@ AbstractQuery.prototype.isUpdateQuery = function() {
 
 AbstractQuery.prototype.handleSelectQuery = function(results) {
   var result = null;
-
+  // Map raw fields to model field names using the `fieldAttributeMap`
+  if (this.options.model && this.options.mapToModel && Utils._.size(this.options.model.fieldAttributeMap) > 0) {
+    var fieldAttributeMap = this.options.model.fieldAttributeMap;
+    results = Utils._.map(results, function(result) {
+      return Utils._.reduce(fieldAttributeMap, function(result, name, field) {
+        if (result[field] !== undefined) {
+          result[name] = result[field];
+          delete result[field];
+        }
+        return result;
+      }, result);
+    });
+  }
```

Comment:
Sorry for being so late to comment.
Ideally i'd like a generic mapper passed to query, or one generated in query and then used here.

The less the query generator and query handler needs to know about `Model` the better.